source/KeyHandlers.js

```diff
@@ -306,9 +306,49 @@ var keyHandlers = {
                 self._updatePath( range, true );
             }
         }
-        // Otherwise, leave to browser but check afterwards whether it has
-        // left behind an empty inline tag.
+        // Nate: previously this was left to the browser but had issues with non-editable spans.  Furthermore
+        // firefox had an odd bug where it is confused by non-editable spans causing spaces to be added
+        // inside the non-editable block rather than deleting the appropriate character inside editable
+        // text.  There is some information on what is probably the same bug here:
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=685445
         else {
```

Comment:
This is a perfect example of a seemingly innocuous task taking a lot more to do than expected... It also seems to produce very nice behavior - very cool!